diff --git a/platform.h b/platform.h
index 751957c..5092078 100644
--- a/platform.h
+++ b/platform.h
@@ -77,9 +77,12 @@
 #elif defined(__arc__)
 	#define __FLASHROM_ARCH__ "arc"
 	#define IS_ARC 1
+#elif defined (__loongarch) || defined (__loongarch__) || defined (__LOONGARCH__) || defined (loongarch)
+	#define __FLASHROM_ARCH__ "loongarch"
+	#define IS_LOONGARCH 1
 #endif
 
-#if !(IS_X86 || IS_MIPS || IS_PPC || IS_ARM || IS_SPARC || IS_ALPHA || IS_HPPA || IS_M68K || IS_RISCV || IS_SH || IS_S390 || IS_ARC)
+#if !(IS_X86 || IS_MIPS || IS_PPC || IS_ARM || IS_SPARC || IS_ALPHA || IS_HPPA || IS_M68K || IS_RISCV || IS_SH || IS_S390 || IS_ARC || IS_LOONGARCH)
 #error Unknown architecture
 #endif
 
